drivers/modem/gsm_ppp.c

```diff
@@ -952,10 +970,15 @@ static void mux_setup(struct k_work *work)
 	/* We need to call this to reactivate mux ISR. Note: This is only called
 	 * after re-initing gsm_ppp.
 	 */
-	if (IS_ENABLED(CONFIG_GSM_MUX) &&
-	    gsm->ppp_dev && gsm->state == STATE_CONTROL_CHANNEL) {
+	if (gsm->ppp_dev && gsm->state == STATE_CONTROL_CHANNEL) {
```

Comment:
As this was the old code of the driver, which I just extended for each of the different mux channels, I do see your point. In the old driver code it was required that the mux channel was enabled again when this function was first entered. Therefore, it checks the state in which the function is at the moment. Nevertheless, this is indeed already a sloppy implementation of the old driver and I will generate a more suitable solution for it now, since I extended the code.